src/extlibs/hrvo/linear_velocity_agent.cpp

```diff
@@ -0,0 +1,66 @@
+#include "linear_velocity_agent.h"
+
+LinearVelocityAgent::LinearVelocityAgent(Simulator *simulator, const Vector2 &position,
+                                         float radius, const Vector2 &velocity,
+                                         float maxSpeed, float maxAccel,
+                                         std::size_t goal_index, float goalRadius)
+    : Agent(simulator, position, radius, velocity, velocity, maxSpeed, maxAccel,
+            goal_index, goalRadius)
+{
+}
+
+void LinearVelocityAgent::computeNewVelocity()
+{
+    pref_velocity_ =
+        simulator_->goals_[goal_index_]->getCurrentGoalPosition() - position_;
+
+    if (abs(pref_velocity_) > max_speed_)
+    {
+        pref_velocity_ = normalize(pref_velocity_) * max_speed_;
+    }
+
+    const float dv = abs(pref_velocity_ - velocity_);
+    if (dv != 0.f)
+    {
+        new_velocity_ = (1.0f - (max_accel_ * simulator_->timeStep_ / dv)) * velocity_ +
+                        (max_accel_ * simulator_->timeStep_ / dv) * pref_velocity_;
```

Comment:
could you comment some details about the calculations happening in this function